packages/api/src/period/controllers.ts

```diff
@@ -217,69 +219,98 @@ const assignQuantifiersDryRun = async (
-        !quantifierAccountIds.includes(r._id.toString()) &&
-        ((q.receivers.length === 0 && r.praiseCount >= praisePerQuantifier) ||
-          assignedPraiseCount(q) + r.praiseCount < maxPraisePerQuantifier)
-      ) {
-        // Assign receiver to quantifier
-        q.receivers.push(r);
-        r.assignedQuantifiers = r.assignedQuantifiers
-          ? r.assignedQuantifiers + 1
-          : 1;
-        // No more assigns needed for this receiver, splice from receiver array
-        if (r.assignedQuantifiers === quantifiersPerPraiseReceiver) {
-          receivers.splice(ri, 1);
-          ri--;
-        }
-      }
+
+  // Assign each bin to an available quantifier
+  let availableQuantifiers = quantifierPool.slice();
+  const availableBins = redundantAssignmentBins.slice();
+
+  const assignmentBinsNeedingQuantifiers: Receiver[][] = [];
+  const availableQuantifiersSkipped: Quantifier[] = [];
+
+  while (availableBins.length > 0) {
+    const assignmentBin: Receiver[] | undefined = availableBins.pop();
+    if (!assignmentBin) continue;
+
+    if (availableQuantifiers.length === 0) {
+      assignmentBinsNeedingQuantifiers.push(assignmentBin);
+      continue;
     }
 
-    // Extend the pool with dummy quantifiers if assigns remain to be done
-    // when reaching the end of the pool
-    if (
-      qi === pool.length - 1 &&
-      assignsRemaining(receivers, quantifiersPerPraiseReceiver)
-    ) {
-      pool.push({
-        _id: undefined,
-        accounts: [],
-        receivers: [],
-      });
+    const q = availableQuantifiers.pop();
+    const qUserAccountIds: string[] = q.accounts.map((account: UserAccountDocument) =>
+      account._id.toString()
+    );
+    const assignmentReceiverIds: string[] = assignmentBin.map((r: Receiver) =>
+      r._id.toString()
+    );
+
+    // Confirm none of the Receivers in the assignment bin belong to the Quantifier
+    const overlappingUserAccounts = intersection(qUserAccountIds, assignmentReceiverIds);
+    if (overlappingUserAccounts.length === 0) {
+      // assign Quantifier to original pool
+      quantifierPoolById[q._id.toString()].receivers.push(...assignmentBin);
+    } else {
+      // make quantifier & praise available for assignment elsewhere
+      availableQuantifiers.unshift(q);
+      availableBins.unshift(assignmentBin);
+
+      // Shuffle order of quantifiers
+      //  or skip if unassignable twice.
+      if (availableQuantifiersSkipped.includes(q)) {
+        assignmentBinsNeedingQuantifiers.push(assignmentBin);
+        continue;
+      } else {
+        availableQuantifiers = shuffle(availableQuantifiers);
+        availableQuantifiersSkipped.push(q);
+      }
     }
   }
 
-  // Trim any unassigned quantifiers from the returned pool
-  while (pool[pool.length - 1].receivers.length === 0) {
-    pool.pop();
+  // Convert object of quantifiers back to array & remove any unassigned
+  const poolAssignments: Quantifier[] = Object.values(quantifierPoolById)
```

Comment:
I'm getting error here. Adding type to Object.values resolves it. 

```
Object.values<Quantifier>
```

Reply:
Added. Was this a typescript error at launch? I'm not seeing it for some reason.

Reply:
nvm i think it was eslint